pkg/transport/limit_listen.go

```diff
@@ -23,8 +23,12 @@ import (
 	"time"
 )
 
+type connMap map[string]chan struct{}
```

Comment:
the map can be made local to the listener, but then the user(auth store in this case) should have access to the instance of the listener to query it.
the key in the map is the client's remote address (ip:port) is that not enough to keep uniqueness, even if the map is global